spec/services/user_signup_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'spec_helper'
+
+describe UserSignup do
+  describe ".perform" do
+    subject(:perform) do
+      described_class.perform(user: user,
+                              invitation_token: invitation_token,
+                              stripe_token: stripe_token)
+    end
+    let(:stripe_token) { "garble" }
+    let(:invitation_token) { nil }
+
+    context "with valid personal information" do
+      let(:user) do
+        Fabricate.build(:user,
+                        fullname: "Joe Doe",
+                        email: 'joe@doe.com')
+      end
+
+      before do
+        expect(StripeWrapper::Charge).to receive(:create).and_return(charge)
+      end
+
+      context "and valid credit card" do
+        let(:charge) { double(:charge, successful?: true) }
+
+        after { ActionMailer::Base.deliveries.clear }
+
+        it "creates new user" do
+          perform
```

Comment:
That's right. I have added a `subject` to each `context` and gotten rid of some unnecessary `let`s.